test/integration/controller/trainjob_controller_test.go

```diff
@@ -33,7 +34,7 @@ import (
 
 	trainer "github.com/kubeflow/trainer/pkg/apis/trainer/v1alpha1"
 	"github.com/kubeflow/trainer/pkg/constants"
-	jobsetplugin "github.com/kubeflow/trainer/pkg/runtime/framework/plugins/jobset"
+	jobsetplgconsts "github.com/kubeflow/trainer/pkg/runtime/framework/plugins/jobset/constants"
 	testingutil "github.com/kubeflow/trainer/pkg/util/testing"
 	"github.com/kubeflow/trainer/test/integration/framework"
 	"github.com/kubeflow/trainer/test/util"
```

Comment:
Should we combine `ginkgo.When("Reconciling TrainJob")` and `ginkgo.Describe("TrainJob controller")` under single Ginkgo context ?
I don't think that we use `trainjob_controller_test.go` file for integration tests outside of TrainJob controller context.
I just think that [this part](https://github.com/kubeflow/trainer/blob/76dff03117d3a191b2c625b116344f980d389686/test/integration/controller/trainjob_controller_test.go#L80-L120) can be easily moved under:
```
ginkgo.Describe("TrainJob controller") 
```

Reply:
We do not want to do it. Ideally, we should decouple framework tests to dedicated `When` (currently, we have all cases in a single `When`).